### PR TITLE
ARTEMIS-1335 Update Netty to 4.1.14

### PR DESCRIPTION
**Pull request by michaelandrepearce**
Update Netty to 4.1.14

### CODE REVIEW

File: artemis-journal/src/main/java/io/netty/buffer/UnpooledUnsafeDirectByteBufWrapper.java [hunk 1]
```diff
@@ -563,7 +564,11 @@ protected SwappedByteBuf newSwappedByteBuf() {
    @Override
    public ByteBuf setZero(int index, int length) {
       if (hasMemoryAddress()) {
-         UnsafeByteBufUtil.setZero(this, addr(index), index, length);
+         if (length == 0) {
```
Comment (michaelandrepearce) [inline artemis-journal/src/main/java/io/netty/buffer/UnpooledUnsafeDirectByteBufWrapper.java]: We have to do this as UnsafeByteBufUtil.setZero(this, addr(index), index, length); util method is removed in 4.1.14 this was the underlying util logic.

**Comment by michaelandrepearce** [comment]
@clebertsuconic this is ready.

File: artemis-journal/src/main/java/io/netty/buffer/UnpooledUnsafeDirectByteBufWrapper.java (hunk 1 above)
Comment (michaelandrepearce) [inline artemis-journal/src/main/java/io/netty/buffer/UnpooledUnsafeDirectByteBufWrapper.java]: @franz1981 if you want to quickly just check.

**Comment by franz1981** [comment]
Seems ok to me! Looking at the code I've written some time ago I probably need to check if every method is inlined or not...some methods look very long!!!

**Comment by michaelandrepearce** [comment]
@franz1981 it is just copy paste of the code that was in the util in previous version before it was removed so code length is the same

**Comment by michaelandrepearce** [comment]
Or do you mean should we make it a static method and then invoke that?

**Comment by franz1981** [comment]
You did already good :) 
I was talking about my previous work on that class

**Comment by michaelandrepearce** [comment]
Ok so happy we merge this, then, i assume. With he jproton and qpid-jms bits merged already, i would be keen for this to make it in, before tagging of 2.3.0.

**Pull request closed without merging**